2435_Paths_in_Matrix_Whose_Sum_Is_Divisible_by_K/try_1.cpp

```diff
@@ -0,0 +1,47 @@
+class Solution {
+public:
+    int numberOfPaths(vector<vector<int>>& grid, int k) {
+        int m = grid.size();
+        int n = grid[0].size();
+        
+        int MOD = k;
+        int M = 1e9 + 7;
+        
+        vector<vector<unordered_map<int, int>>> dp(m, vector<unordered_map<int, int>>(n, unordered_map<int, int>{}));
+        
+        dp[0][0][grid[0][0] % MOD] = 1;
+        for (int i=1; i<m; i++) {
+            int val = grid[i][0];
+            for (auto &item : dp[i-1][0]) {
```

Comment:
```suggestion
            for (auto &[k, v] : dp[i-1][0]) {
```
unordered_map可以這樣取對應key value

Reply:
Done